src/commands/index.rs

```diff
@@ -145,18 +215,29 @@ where
         .ok_or("ERR no such index".into())
         .map(|schema| &schema.index)
         .and_then(|index| {
-            let results: Result<Vec<_>, RedisError> = index
-                .search(&query)?
-                .map(|key| {
-                    let key = ctx.open_key_writable(&key);
-                    let value = match key.get_value::<RedisJSON>(&REDIS_JSON_TYPE)? {
-                        Some(doc) => doc.to_string(&path, Format::JSON)?.into(),
-                        None => RedisValue::None,
-                    };
-                    Ok(value)
-                })
-                .collect();
-
-            Ok(results?.into())
+            let result: Value =
+                index
+                    .search(&query)?
+                    .try_fold(Value::Object(Map::new()), |mut acc, key| {
+                        ctx.open_key(&key)
+                            .get_value::<RedisJSON>(&REDIS_JSON_TYPE)
+                            .and_then(|doc| {
+                                doc.map_or(Ok(Vec::new()), |data| {
+                                    data.get_values(&path)
+                                        .map_err(|e| e.into()) // Convert Error to RedisError
+                                        .map(|values| {
+                                            values.iter().map(|val| (*val).clone()).collect()
```

Comment:
Instead of using `iter` and then dereferencing `val` manually, you can do this:
```
values.into_iter().map(|val| val.clone()).collect()
```

Reply:
done